### PR TITLE
Add Tutorial for AgentChat Docs

### PR DESCRIPTION
**Pull request by victordibia**
## Why are these changes needed?

Add Tutorial to AgentChat Docs 

image

## Related issue number

- #3828 (adds a non code quickstart) 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by victordibia** [comment]
@ekzhu , happy to get your thoughts.

- If it makes sense we can remove the current tool_use, code_execution notebooks as they are covered in the tutorial. 
- Also happy to get your thoughts on how `on_messages` is introduced as a way to "test" agent behavior in /tutorials/agents. 
- Thoughts on the right way to test code execution agent with the `on_messages` method.  

image

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+myst:
+  html_meta:
+    "description lang=en": |
+      Tutorial for AutoGen AgentChat, a framework for building multi-agent applications with AI agents.
+---
+
```
Comment (ekzhu) [inline python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/index.md:7]: We need a WIP notice here to make sure people understand this is a work in progress

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb [hunk 2]
```diff
@@ -0,0 +1,328 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "# Agents\n",
```
Comment (ekzhu) [inline python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb:9]: Need a WIP warning here to indicate the current status of the API.

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/index.md (hunk 1 above)
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/index.md:7]: Added a warning to all top level pages (index, quickstart, tutorial)

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/index.md [hunk 3]
```diff
@@ -16,26 +16,40 @@ If you are interested in implementing complex agent interaction behaviours, defi
 
 ```
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/index.md:18]: Added a warning to all top level pages (index, quickstart, tutorial)

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    "\n",
+    "An agent is a software entity that communicates via messages, maintains its own state, and performs actions in response to received messages or changes in its state. \n",
+    "\n",
+    "AgentChat provides a set of preset Agents, each with variations in how an agent might respond to received messages. \n",
+    "\n",
+    "Each agent inherits from a `BaseChatAgent` class with a few generic properties:\n",
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb:15]: Thanks, added

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    "\n",
+    "AgentChat is built on the `autogen-core` package, which handles the details of sending and receiving messages. `autogen-core` provides a runtime environment, which facilitates communication between agents (message sending and delivery), manages their identities and lifecycles, and enforces security and privacy boundaries. \n",
+    "AgentChat handles the details of instantiating a runtime and registering agents with the runtime.\n",
+    "\n",
+    "To learn more about the runtime in `autogen-core`, see the [autogen-core documentation on agents and runtime](../../core-user-guide/framework/agent-and-agent-runtime.ipynb).\n",
+    "```\n",
+    "\n",
+    "Each agent also implements an `on_messages` method that defines the behavior of the agent in response to a message.\n",
+    "\n",
+    "## CodingAssistantAgent\n",
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb]: Removed this for now. (commented out) until the api is updated

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    "```{tip}\n",
+    "How do agents send and receive messages? \n",
+    "\n",
+    "AgentChat is built on the `autogen-core` package, which handles the details of sending and receiving messages. `autogen-core` provides a runtime environment, which facilitates communication between agents (message sending and delivery), manages their identities and lifecycles, and enforces security and privacy boundaries. \n",
+    "AgentChat handles the details of instantiating a runtime and registering agents with the runtime.\n",
+    "\n",
+    "To learn more about the runtime in `autogen-core`, see the [autogen-core documentation on agents and runtime](../../core-user-guide/framework/agent-and-agent-runtime.ipynb).\n",
+    "```\n",
+    "\n",
+    "Each agent also implements an `on_messages` method that defines the behavior of the agent in response to a message.\n",
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb]: Thanks, added

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb (hunk 2 above)
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/agents.ipynb:9]: Added

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/teams.ipynb [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly, we can define a team where the agents solve a problem by _writing and executing code_ in a round-robin fashion. \n",
+    "\n",
+    "```python \n",
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/teams.ipynb:117]: There is a note there that explictly tells the user that Docker needs to be installed and running, encouraging them to skip that section if they do not have docker installed.

File: python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/installation.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+`````{tab-item} poetry
+
+[Install Poetry](https://python-poetry.org/docs/#installation) if you have not already.
+
+Create and activate:
+```bash
+poetry init
+poetry shell
+
+poetry add autogen-agentchat~=0.2
```
[the remark this answers is not in the document]
Comment (victordibia) [reply python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/installation.md]: Updated, removed poetry tab

**Review by ekzhu — APPROVED** [review]

**Pull request merged**